### DISCUSSION TITLE
setting exact time for polling

### DISCUSSION THREAD

**Discussion by bsulyok** [elkowar/eww#738] [General]
I am using two distinct widgets for time and date, I am updating the time widget every second, but I want to update the date widget only every 24 hours, specifically at 00:00 every day, can I do this?

**Comment by viandoxdev** [comment]
There is no eww builtin that allows for that, but you could do either:
- some kind of daily cron job running `eww update var=...`
- just poll every minute, while this does waste some performance, the cost of running a single `date` command isn't really anything you'll ever notice (unless you're somehow running eww on a low power embeded platform)

Note that even if there was a way to do that, it wouldn't be much faster anyways, since it would probably be polling the time anyways (to check when to run)

**Comment by oldwomanjosiah** [comment]
I would recommend writing a bash script like

```bash
#!/usr/bin/env bash

# Log to eww (stderr) instead of sending data (stdout)
function log() {
    >&2 echo "$@"
}

# Fetches data that needs to be updated daily
function doOutput() {
    log "Getting output"
    local dailyData=$(getSomeData)
    echo "{ \"dailyData\": \"$dailyData\" }"
}

# Wait until specified time
# See https://unix.stackexchange.com/questions/259660/how-to-sleep-until-a-given-date-time
function waitForTime() {
    local targetTime="tomorrow 00:00"
    
    local now=$(date +%s)
    local sleepTill=$(date +%s -d "$targetTime")
    local sleepTime=$(expr $sleepTill - $now)    

    log "Sleeping for $sleepTime seconds (until $targetTime)"
    sleep sleepTime
}

# Output once so that we get info whenever eww starts up
doOutput

while 1; do
  waitForTime
  doOutput
done
```

which you can then use as

```clojure
(deflisten someData
  :initial "{}"
  `./path/to/script`)

(defwidget usesSomeData (...))
```
